File: angular/src/app/services/File/dtos/CreateFileInput.ts
import { HomeworkFullOutPut } from './../../Homework/dtos/HomeworkFullOutPut';
export class CreateFileInput implements ICreateFileInput{
    files: File[];
    homework: HomeworkFullOutPut;
}

export interface ICreateFileInput{
    files:File[]|undefined;
    homework:HomeworkFullOutPut|undefined;
}

File: aspnet-core/src/Lesson.Application/Domain/File/Dto/CreateFileInput.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lesson.Domain.File.Dto
{
    public class CreateFileInput
    {
        public List<IFormFile> Files { get; set; }
        public Homework.Dto.HomeworkFullOutPut Homework { get; set; }
    }
}
